### PR TITLE
chore: move to noble

### PR DESCRIPTION
**Pull request by hpidcock**
This is a whole bunch of fun in one ugly PR.
- update jenkins job builder to 6.3.0
- fix all the issues around parameters vs templates
- remove some legacy files
- add a nix flake for development shell (mostly for me)
- add move to noble

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 
-cwd 			 = $(shell pwd)
-virtualenv_dir 	 = $(cwd)/venv
+cwd              = $(shell pwd)
+virtualenv_dir   = $(cwd)/venv
 python_base_path = $(shell which python3)
 
 .PHONY: ensure-venv
 ensure-venv:
-	pip3 show virtualenv > /dev/null || pip3 install virtualenv
```
Comment (wallyworld) [inline Makefile:14]: We don't want this anymore?

**Review by wallyworld — APPROVED** [review]
Awesome. \o/

File: Makefile (hunk 1 above)
Comment (SimonRichardson) [reply Makefile:14]: Nope.

Comment (hpidcock) [reply Makefile:14]: python has built in venv module

**Pull request merged**